src/ln/channelmanager.rs

```diff
@@ -262,6 +262,12 @@ pub(super) struct ChannelHolder {
 	/// Messages to send to peers - pushed to in the same lock that they are generated in (except
 	/// for broadcast messages, where ordering isn't as strict).
 	pub(super) pending_msg_events: Vec<events::MessageSendEvent>,
+	/// Due to reorgs, we want to be sure that commitment txn has enough depth
+	/// (HTLC_FAIL_ANTI_REORG_DELAY).
+	/// before to close channel. Doing it to early would block us to pass backward
+	/// preimage in case of successful claim on onward channel. Key is block
+	/// height timer, value is array of channels id, waiting their closing.
+	pub(super) channel_closing_waiting_threshold_conf: HashMap<u32, Vec<[u8; 32]>>,
```

Comment:
Wait, why? If our counterparty broadcasts a commitment transaction, and it gets reorged out, we shoulnd't try to relay payments over the channel, that commitment transaction is probably gonna be mined again in a block soon, we should just move forward happily with a closed channel.

Reply:
Oh I'm dumb, that's exactly what my test was trying to do, i.e testing some edge cases which is doable but doesn't make sense on a high-level. And from then, adding all this non-sense struct in channel_manager, I've dropped this commit and rewrote test_sweep_outbound_htlc_faiilure_update.